gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -294,10 +335,30 @@ public synchronized void setActive(boolean active) {
     @Override
     public void run() {
       try {
-        Object nextItem = queue.poll();
+        String nextDagToCancel = cancelQueue.poll();
+        //Poll the cancelQueue for a new Dag to cancel.
+        if (nextDagToCancel != null) {
+          log.info("Cancel flow with DagId {}", nextDagToCancel);
+          if (this.dagToJobs.containsKey(nextDagToCancel)) {
+            List<DagNode<JobExecutionPlan>> dagNodesToCancel = this.dagToJobs.get(nextDagToCancel);
+            log.info("Found {} DagNodes to cancel.", dagNodesToCancel.size());
+            for (DagNode<JobExecutionPlan> dagNodeToCancel : dagNodesToCancel) {
+              Properties props = new Properties();
+              if (dagNodeToCancel.getValue().getAzkabanExecId().isPresent()) {
+                String azkabanExecId = dagNodeToCancel.getValue().getAzkabanExecId().get().getExecId();
+                props.put(ConfigurationKeys.AZKABAN_EXEC_ID, azkabanExecId);
+                log.info("Cancel job with azkaban exec id {}.", azkabanExecId);
+              }
+              DagManagerUtils.getSpecProducer(dagNodeToCancel).deleteSpec(null, props);
```

Comment:
Is deleting a spec enough ? Does this need to involve cancelling of ongoing azkaban execution ?

Reply:
AzkabanHeadlessSpecProducer.delete() actually cancels the flow